File: dotnet-desktop-guide/net/winforms/controls/snippets/how-to-make-thread-safe-calls/cs/FormBad.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace project
{
    public partial class FormBad : Form
    {
        public FormBad()
        {
            InitializeComponent();
        }

        //<Bad>
        private void button1_Click(object sender, EventArgs e)
        {
            var thread2 = new System.Threading.Thread(WriteTextUnsafe);
            thread2.Start();
        }

        private void WriteTextUnsafe() =>
            textBox1.Text = "This text was set unsafely.";
        //</Bad>
    }
}

File: dotnet-desktop-guide/winforms/controls/snippets/how-to-make-thread-safe-calls/vb/FormBad.vb
Public Class FormBad
    '<Bad>
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim thread2 As New System.Threading.Thread(AddressOf WriteTextUnsafe)
        thread2.Start()
    End Sub

    Private Sub WriteTextUnsafe()
        TextBox1.Text = "This text was set unsafely."
    End Sub
    '</Bad>
End Class
